### PR TITLE
#278: Hide "Edit" button when all addresses are deleted

### PR DESCRIPTION
**Pull request by ahmed-shehata**
#278

### CODE REVIEW

File: AtlasUI/AtlasUI/AddressListTableViewDelegate.swift [hunk 1]
```diff
@@ -86,6 +86,17 @@ extension AddressListTableViewDelegate: UITableViewDelegate {
...
 
+    private func deleteAddress(indexPath: NSIndexPath, tableView: UITableView) {
+        self.addresses.removeAtIndex(indexPath.item)
+
+        if self.addresses.isEmpty {
+            tableView.setEditing(true, animated: true)
+        }
+
+        self.deleteAddressHandler?()
+        tableView.deleteRowsAtIndexPaths([indexPath], withRowAnimation: .Fade)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/AddressListTableViewDelegate.swift]: Maybe `animation: .automatic`?

File: AtlasUI/AtlasUI/AddressPickerViewController.swift [hunk 2]
```diff
@@ -181,7 +184,7 @@ extension AddressPickerViewController {
 extension AddressPickerViewController {
 
     private func configureUpdateAddress() {
-        tableviewDelegate?.updateAddressHandler = { [weak self] (address) in
+        tableviewDelegate?.updateAddressHandler = { [weak self](address) in
```
Comment (bonkey) [inline AtlasUI/AtlasUI/AddressPickerViewController.swift]: style: `[weak self] address in`

File: AtlasUI/AtlasUI/AddressPickerViewController.swift [hunk 3]
```diff
@@ -215,3 +218,21 @@ extension AddressPickerViewController {
...
+    }
+
+    private func configureEditButton() {
+        if let addressList = tableviewDelegate?.addresses where addressList.isEmpty {
+            self.navigationItem.rightBarButtonItem = nil
+        } else {
+            self.navigationItem.rightBarButtonItem = self.editButtonItem()
+        }
+    }
+}
```
Comment (bonkey) [inline AtlasUI/AtlasUI/AddressPickerViewController.swift]: Where's this swiftlint rule to keep files 100 lines long? 👹

Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AddressPickerViewController.swift]: NOOOOO 😕 
no more rules :(

Comment (bonkey) [inline AtlasUI/AtlasUI/AddressPickerViewController.swift]: hahaha. it's for your own good ;-)

File: AtlasUI/AtlasUI/AddressPickerViewController.swift (hunk 2 above)
Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/AddressPickerViewController.swift]: swimat keeps deleting it

**Comment by ahmed-shehata** [comment]
👍

**Comment by bonkey** [comment]
👍

**Comment by ahmed-shehata** [comment]
👍

**Comment by bonkey** [comment]
👍

**Comment by bonkey** [comment]
👍

**Comment by artdaw** [comment]
👍

**Pull request merged**